src/transformers/models/luke/tokenization_luke.py

```diff
@@ -0,0 +1,1413 @@
+# coding=utf-8
+# Copyright Studio-Ouisa and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for LUKE."""
+import itertools
+import json
+import os
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+
+from ... import RobertaTokenizer
+from ...file_utils import add_end_docstrings, is_tf_available, is_torch_available
+from ...tokenization_utils_base import (
+    ENCODE_KWARGS_DOCSTRING,
+    ENCODE_PLUS_ADDITIONAL_KWARGS_DOCSTRING,
+    AddedToken,
+    BatchEncoding,
+    EncodedInput,
+    PaddingStrategy,
+    TensorType,
+    TextInput,
+    TextInputPair,
+    TruncationStrategy,
+    _is_tensorflow,
+    _is_torch,
+    to_py_obj,
+)
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {
+    "vocab_file": "vocab.json",
+    "merges_file": "merges.txt",
+    "entity_vocab_file": "entity_vocab.json",
+}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "studio-ousia/luke-base": "https://huggingface.co/studio-ousia/luke-base/resolve/main/vocab.json",
+        "studio-ousia/luke-large": "https://huggingface.co/studio-ousia/luke-large/resolve/main/vocab.json",
+    },
+    "merges_file": {
+        "studio-ousia/luke-base": "https://huggingface.co/studio-ousia/luke-base/resolve/main/merges.txt",
+        "studio-ousia/luke-large": "https://huggingface.co/studio-ousia/luke-large/resolve/main/merges.txt",
+    },
+    "entity_vocab_file": {
+        "studio-ousia/luke-base": "https://huggingface.co/studio-ousia/luke-base/resolve/main/entity_vocab.json",
+        "studio-ousia/luke-large": "https://huggingface.co/studio-ousia/luke-large/resolve/main/entity_vocab.json",
+    },
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "studio-ousia/luke-base": 512,
+    "studio-ousia/luke-large": 512,
+}
+
+
+class LukeTokenizer(RobertaTokenizer):
```

Comment:
If we are overriding so many methods would be nice to not inherit from `RobertaTokenizer`, AFAIK we don't inherit tokenizers anymore.

cc @LysandreJik , @sgugger

Reply:
Inheriting from tokenizers is actually acceptable in most cases - here we are indeed overriding a lot of methods, but a lot of methods relative to the actual tokenization still come from the RoBERTa (and, therefore, the GPT-2) tokenizer, so I personally have nothing about keeping the inheritance as it is.

If you or anybody else feels strongly, happy to have this conversation and see what's our objective with tokenizers as it isn't aligned with the models (which is fine to me).

Reply:
Yes, let's open the discussion on that as I'm not super happy about the inheritance either when there is a lot of stuff rewritten :-) 

(It's fine to leave as is in this PR while we debate this internally.)